Material: Fix selected tab indicator animation.

mpcomplete opened the pull request:
- The indicator no longer goes crazy when you select another tab while
  it's animating.
- The indicator's trailing edge lags behind the trailing edge, giving it
  a neat stretchy feel.

BUG=https://github.com/flutter/flutter/issues/1617
BUG=https://github.com/flutter/flutter/issues/1955

Hixie commented on packages/flutter/lib/src/material/tabs.dart:
```diff
@@ -660,11 +666,27 @@ class _TabBarState<T> extends ScrollableState<TabBar<T>> implements TabBarSelect
-    // might not have set up our _indicatorTween yet.
-    _indicatorRect = _indicatorTween.lerp(t);
     if (oldRect != _indicatorRect)
       setState(() { });
```
while you're here, please put a comment in the block here saying what it is that changed

Hixie commented:
lgtm but @HansMuller really should review this

mpcomplete committed:
Material: Fix selected tab indicator animation.

- The indicator no longer goes crazy when you select another tab while
  it's animating.
- The indicator's trailing edge lags behind the trailing edge, giving it
  a neat stretchy feel.

BUG=https://github.com/flutter/flutter/issues/1617
BUG=https://github.com/flutter/flutter/issues/1955

HansMuller commented:
LGTM
